src/test/test-execute.c

```diff
@@ -28,8 +30,41 @@
 #include "util.h"
 #include "virt.h"
 
+static int status_may_namespace = 0;
+static int status_may_fail = 0;
+
 typedef void (*test_function_t)(Manager *m);
 
+static void check_mount_namespace(void) {
+        const char *name = "(setup_namespace)";
+        pid_t pid;
+        int r;
+
+        log_info("Testing mount namespace");
+
+        r = safe_fork(name, FORK_DEATHSIG|FORK_LOG, &pid);
+        assert_se(r >= 0);
+
+        if (r == 0) { /* Child */
+                NamespaceInfo ns_info = {};
+                r = setup_namespace(NULL, NULL, &ns_info, NULL, NULL, NULL, NULL, NULL, 0, NULL, 0, NULL, NULL, PROTECT_HOME_NO, PROTECT_SYSTEM_YES, 0, 0);
+                if (r < 0)
+                        _exit(-r);
+
+                _exit(EXIT_SUCCESS);
+        }
+
+        /* Parent */
+        r = wait_for_terminate_and_check(name, pid, WAIT_LOG);
+        assert_se(r >= 0);
+
+        if (r == ENOANO) {
```

Comment:
Maybe move the ENOANO check into the child process? Then we don't have to worry about whether `ENOANO > 0xff`.

Reply:
Thank you for the comment. I will update this accordingly.

Reply:
I use the following mapping in the revised version:
- `r >= 0` -> `EXIT_SUCCESS`
- `r == -ENOANO` -> `EXIT_NAMESPACE`
- `r < 0` -> `EXIT_FAILURE`